pydantic/generics.py

```diff
@@ -38,6 +38,8 @@
 # `Model[int, str]`: {A: int, B: str}` will be stored in `_assigned_parameters`.
 # (This information is only otherwise available after creation from the class name string).
 _assigned_parameters: Dict[Type[Any], Parametrization] = {}
+# _assigned_parameters size is limited by _limit_assigned_parameters() below
+_assigned_parameters_max_length = 1000
```

Comment:
If `_assigned_parameters` is a kind of cache we can limit in size, could we instead also use `lru_cache` for `__parameterized_bases__`?:
```
@classmethod
@lru_cache(maxsize=1024)
def __parameterized_bases__(cls, typevars_map: Parametrization) -> List[Type[Any]]:
```
And then remove the `_assigned_parameters` "cache"?

Reply:
Yes we could, in fact that's how I had it on the first commit. But I wanted to keep the interface identical (in case people are introspecting `_assigned_parameters`) and also allow people to change the limit. 

Hence this approach.

Reply:
Sorry, I realise I never pushed the second commit, should make more sense now.